domain/domain.go

```diff
@@ -942,32 +968,20 @@ func (do *Domain) Init(
 			return err
 		}
 		if addrs != nil {
-			cfg := config.GetGlobalConfig()
-			// silence etcd warn log, when domain closed, it won't randomly print warn log
-			// see details at the issue https://github.com/pingcap/tidb/issues/15479
-			etcdLogCfg := zap.NewProductionConfig()
-			etcdLogCfg.Level = zap.NewAtomicLevelAt(zap.ErrorLevel)
-			cli, err := clientv3.New(clientv3.Config{
-				LogConfig:        &etcdLogCfg,
-				Endpoints:        addrs,
-				AutoSyncInterval: 30 * time.Second,
-				DialTimeout:      5 * time.Second,
-				DialOptions: []grpc.DialOption{
-					grpc.WithBackoffMaxDelay(time.Second * 3),
-					grpc.WithKeepaliveParams(keepalive.ClientParameters{
-						Time:    time.Duration(cfg.TiKVClient.GrpcKeepAliveTime) * time.Second,
-						Timeout: time.Duration(cfg.TiKVClient.GrpcKeepAliveTimeout) * time.Second,
-					}),
-				},
-				TLS: ebd.TLSConfig(),
-			})
+			cli, err := newEtcdCli(addrs, ebd)
 			if err != nil {
 				return errors.Trace(err)
 			}
 
 			etcd.SetEtcdCliByNamespace(cli, keyspace.MakeKeyspaceEtcdNamespace(do.store.GetCodec()))
 
 			do.etcdClient = cli
+
+			noNamespaceCli, err := newEtcdCli(addrs, ebd)
+			if err != nil {
+				return errors.Trace(err)
+			}
+			do.unprefixedEtcdCli = noNamespaceCli
```

Comment:
If there isn't set `keyspace-name`, we will not set etcd namespace for etcdClient. All the logic about `etcdClient` is the same as the old logic.
If we set `keyspace-name`, only the etcd path for `minstartts` does not need to add the etcd namespace, other etcd paths must add the etcd namespace prefix to separate by keyspace.
So they need exist at the same time.

How to `unprefixedEtcdCli` or `etcdClient`?
The `etcdClient` must be used when the logic to each etcd path needs to be executed separately by keyspace.
This `unprefixedEtcdCli` must be used when the logic to each etcd path needs to be used globally.

Reply:
It's still confusing, consider if someone is going to add a new feature and an `etcd client` would be used to do something, which one should be used so that the keyspace or non-keyspace constraints would not be broken?
Maybe detailed explanations and remindings in the comments are needed or a link to the RFC.

Reply:
I have added comments and the RFC link.PTAL~